02-peer-protocol.md

```diff
@@ -388,17 +388,7 @@ the byte size of the input and output counts on the transaction to one (1).
     * [`channel_id`:`channel_id`]
     * [`sha256`:`txid`]
     * [`u16`:`num_witnesses`]
-    * [`num_witnesses*witness_stack`:`witnesses`]
-
-1. subtype: `witness_stack`
-2. data:
-    * [`u16`:`num_witness_elements`]
-    * [`num_witness_elements*witness_element`:`witness_elements`]
-
-1. subtype: `witness_element`
-2. data:
-    * [`u16`:`len`]
-    * [`len*byte`:`witness_data`]
+    * [`num_witnesses*witness`:`witnesses`]
```

Comment:
I liked Rusty's suggestion better:

```suggestion
    * [`num_witnesses*witness`:`witnesses`]
    
1. subtype: `witness`
2. data:
    * [`u16`:`len`]
    * [`len*byte`:`witness`]
```

Otherwise with your current proposal, lightning implementations are forced to fully decode every `witness` to be able to split the tlv into separate `witness`es and add them to their corresponding `TxIn`. I thought that was what we wanted to avoid!

Rusty's proposal adds a somewhat redundant `u16` length (since this length would be inferred by fully decoding the `witnesses`), but it's also more consistent with the rest of the lightning spec. I don't think we care about a few bytes wasted if it improves overall consistency and simplifies our decoding logic.